docs/resources/kubernetes_secret.md

```diff
@@ -0,0 +1,138 @@
+
+In managed clusters, both attached and provisioned, you can create Kubernetes Secret that you can manage through Tanzu Mission Control.
+
+To create a cluster secret, you must be associated with the cluster.admin role
+
+The `tanzu-mission-control_kubernetes_secret` resource enables you to create cluster secret to a particular scope for management through Tanzu Mission Control.
+
+You can optionally make export to true to make the secret available to all namespaces in the cluster.
+
+[cluster Secret]: https://docs.vmware.com/en/VMware-Tanzu-Mission-Control/services/tanzumc-using/GUID-BBE2404D-C2EE-41C7-B639-C0322783A74D.html
+
+[export secret to all namespaces]: https://docs.vmware.com/en/VMware-Tanzu-Mission-Control/services/tanzumc-using/GUID-B0A72F72-4216-4869-B293-6802368B11D2.html
+
+## Policy Scope and Inheritance
+
+In the Tanzu Mission Control resource hierarchy, there is currently one levels at which you can specify cluster secret resources:
+- **Kubernetes objects** - `cluster` block under `scope` sub-resource   
+
+**Note:**
+The scope parameter is mandatory in the schema and the user needs to add one of the defined scopes to the script for the provider to function.
+Only one scope per resource is allowed.
+
+## Cluster scoped cluster secret
+
+## Example Usage
+
+```terraform
+# Create Tanzu Mission Control kubernetes secret with attached set as default value.
+resource "tanzu-mission-control_kubernetes_secret" "create_secret" {
+  name                    = "tf-secret" # Required
+  namespace_name          = "tf-secret-namespace-name" # Required 
+
+  scope {
+    cluster {
+        cluster_name            = "testcluster"  # Required
+        provisioner_name        = "attached"     # Default: attached
+        management_cluster_name = "attached"     # Default: attached
+    }
+  }
+
+   export = false # Default: false
+
+  meta {
+    description = "Create namespace through terraform"
+    labels      = { "key" : "value" }
+  }
+
+  spec {
+    docker_config_json {
+        username           = "testusername" # Required
+        password           = "testpassword" # Required
+        image_registry_url = "testimageregistryurl" # Required
+    }
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) Name of the secret resource.
+- `namespace_name` (String) Name of Namespace where secret will be created.
+- `scope` (Block List, Min: 1, Max: 1) Scope for the secret having one of the valid scopes for secret: currently we have only cluster scope (see [below for nested schema](#nestedblock--scope))
+
+### Optional
+
+- `export` (Boolean) Export the secret to all namespaces.
+- `meta` (Block List, Max: 1) Metadata for the resource (see [below for nested schema](#nestedblock--meta))
+- `org_id` (String) ID of Organization.
+- `spec` (Block List, Max: 1) Spec for the kubernetes secret (see [below for nested schema](#nestedblock--spec))
```

Comment:
Is Spec optional? Shouldn't it be required?

Reply:
for data source[Get request for cluster-secret] and resource[Create, delete, update for cluster-secret] we have used the same schema and for data source we don't need spec part to mention, only for resource we need spec so that's why I have made it to Optional.

@ishangupta-ds , can you also take a look

Reply:
@shreyassreenivas do we follow the same for other resources which have corresponding data sources? If yes then can we keep the spec option for the `kubernetes_secret` resource as well? Cc: @Vasundharashukla @narahari92

Reply:
@narahari92 Please take a look, we have fixed the docs and Schema 
**spec** is a required block for Resource and Read only for Data Source.

cc: @ishangupta-ds, @Vasundharashukla , @shreyassreenivas

Reply:
LGTM

Reply:
Resolving as per the discussion with @narahari92